devito/arch/archinfo.py

```diff
@@ -234,7 +257,20 @@ def homogenise_gpus(gpu_infos):
     # Obtain textual gpu info and delegate parsing to helper functions
 
     try:
-        # First try is with detailed command lshw
+        # First try is with nvidia-smi, which can have more info about NVIDIA cards
```

Comment:
Put here because if nvidia-smi is present, we have greater chances of get proper information. Depending on how the driver was installed, lshw and lspci can't retrieve product information.

Reply:
yeah, I suppose if we have an AMD card instead this check will just quickly fail, and the execution will proceed smoothly with `lshw` right?

Reply:
Yes